jetstream/cli.py

```diff
@@ -56,28 +54,71 @@ class AllType:
 
 class ExecutorStrategy(Protocol):
     project_id: str
-    dataset_id: str
 
-    def __init__(self, project_id: str, dataset_id: str, *args, **kwargs) -> None:
+    def __init__(self, project_id: str, *args, **kwargs) -> None:
         ...
 
-    def execute(self, worklist: Iterable[Tuple[str, AnalysisSpec, datetime]]) -> bool:
+    def execute(
+        self,
+        worklist: Iterable[Tuple[str, datetime]],
+        configuration_map: Mapping[str, TextIO] = {},
+    ) -> bool:
         ...
 
 
+@attr.s(auto_attribs=True)
+class ArgoExecutorStrategy:
+    project_id: str
+    zone: str
+    cluster_id: str
+    monitor_status: bool
+    experiment_getter: Callable[[], ExperimentCollection] = ExperimentCollection.from_experimenter
+
+    WORKLFOW_DIR = Path(__file__).parent / "workflows"
+    RUN_WORKFLOW = WORKLFOW_DIR / "run.yaml"
+
+    def execute(self, worklist, _configuration_map: Mapping[str, TextIO] = {}):
+        experiments_config = [
+            {"date": date.strftime("%Y-%m-%d"), "slug": slug} for (slug, date) in worklist
+        ]
+
+        return submit_workflow(
+            project_id=self.project_id,
+            zone=self.zone,
+            cluster_id=self.cluster_id,
+            workflow_file=self.RUN_WORKFLOW,
+            parameters={"experiments": experiments_config},
+            monitor_status=self.monitor_status,
+        )
+
+
 @attr.s(auto_attribs=True)
 class SerialExecutorStrategy:
     project_id: str
     dataset_id: str
     analysis_class: Type = Analysis
     experiment_getter: Callable[[], ExperimentCollection] = ExperimentCollection.from_experimenter
+    config_getter: Callable[
+        [], ExternalConfigCollection
+    ] = ExternalConfigCollection.from_github_repo
 
-    def execute(self, worklist):
+    def execute(
+        self, worklist: List[Tuple[str, datetime]], configuration_map: Mapping[str, TextIO] = {}
+    ):
         failed = False
         experiments = self.experiment_getter()
-        for slug, spec, date in worklist:
+        for slug, date in worklist:
             try:
                 experiment = experiments.with_slug(slug).experiments[0]
+                spec = AnalysisSpec.default_for_experiment(experiment)
+                if slug in configuration_map:
+                    config_dict = toml.load(configuration_map[slug])
+                    spec.merge(AnalysisSpec.from_dict(config_dict))
+                else:
+                    external_configs = self.config_getter()
+                    if external_spec := external_configs.spec_for_experiment(slug):
+                        spec.merge(external_spec)
+
```

Comment:
I moved this in here since the analysis spec is only evaluated when using `SerialExecutorStrategy`

Reply:
I had originally pushed this logic into the CliExecutor so that it could hand the ExecutorStrategy a complete, serializable spec that didn't need access to local resources to evaluate, since I noticed you were passing base64'd JSON around -- I'm not sure how to handle the configuration_map otherwise.

It's possible we just don't accept a configuration_map in Argo mode, which seems fine to me.

Reply:
Yeah, I think it's better to not pass the spec around. The base64 encoded "solution" felt a little hacky, instead we'll just resolve the spec for each experiment right before the analysis.

Reply:
For now, I'd just ignore the configuration map when using Argo. It's anyway not meant to be exposed to users, so I don't think there is an immediate need for having it.